src/MoBi.Presentation/Tasks/Interaction/InteractionTasksForContainerBase.cs

```diff
@@ -121,6 +127,48 @@ private ICommand addParameterValues(ParameterValuesBuildingBlock parameterValues
          if (parameterValues == null || !parameterValues.Any())
             return new MoBiEmptyCommand();
 
+         var parameterValuesBuildingBlock = buildingBlockToAdd(module);
+
+         if (parameterValuesBuildingBlock != null)
+            return addToExistingBuildingBlock(parameterValues.ToList(), parameterValuesBuildingBlock);
+
+         return addAsNewBuildingBlock(parameterValues, module);
+      }
+
+      private ICommand addToExistingBuildingBlock(IReadOnlyList<ParameterValue> parameterValuesToAdd, ParameterValuesBuildingBlock parameterValuesBuildingBlock)
+      {
+         return _parameterValuesTask.ExtendBuildingBlockWith(parameterValuesBuildingBlock, parameterValuesToAdd);
+      }
+
+      private ParameterValuesBuildingBlock buildingBlockToAdd(Module module)
+      {
+         var moduleBuildingBlocks = module.ParameterValuesCollection;
+
+         // If there are no existing building blocks, then you can only add as a new building block
+         if (!moduleBuildingBlocks.Any())
+            return null;
+
+         using (var modal = ApplicationController.Start<IModalPresenter>())
+         {
+            var presenter = ApplicationController.Start<ISelectSinglePresenter<ParameterValuesBuildingBlock>>();
+            presenter.SetDescription(AppConstants.Captions.SelectTheBuildingBlockWhereParameterValuesWillBeAddedOrUpdated);
+            modal.Text = AppConstants.Captions.SelectParameterValuesBuildingBlock;
+            modal.Encapsulate(presenter);
+            
+            presenter.InitializeWith(new List<ParameterValuesBuildingBlock>
```

Comment:
I think this should be rewritten
 var toto = new List<PVV>(moduleBuildingBlockks){NullPathAndValueEntityBuildingBlocks.NewParameterValues}

Or is this not correct?

Reply:
The goal is to have the `NewParameterValues` at the top of the list. I have this other comment from you about the `OrderBy`, so I'm thinking that the presenter could maybe default `OrderBy`. Then here, where I am adding this special item, I can place it at the top somehow.

Reply:
oh so the order is important? Then forget it/ But I think a comment would help to understand why the syntax is unexpected

Reply:
Yes, order is important, but I got to thinking that maybe it's not important for the SelectManyPresenter, but only for the particular caller.

The order is only important in this case because of this special building block that should be at the top.

Anyway, it's been replaced with something that allows the caller to modify the order while initializing.